src/sub/vectors.sub

```diff
@@ -1,5 +1,22 @@
 VectorSpace U
 Vector(U) A
 Vector(U) B
+Vector(U) C
+
+Scalar C1
+ZeroScalar C2
```

Comment:
why `ZeroScalar`?

Reply:
Oh, that is just for my checks of subtyping, we do not need that in the real book....